File: Backend/Backend/Dtos/NotificationDto.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Backend.Dtos
{
    // DTO for creating a new notification
    public class CreateNotificationDto
    {
        [Required]
        public string RecipientId { get; set; }

        [Required]
        public string Message { get; set; }

        public string Type { get; set; }

        public bool IsRead { get; set; } = false;
    }

    // DTO for displaying a notification
    public class NotificationDto
    {
        public string Id { get; set; }

        public string RecipientId { get; set; }

        public string Message { get; set; }

        public DateTime CreatedAt { get; set; }

        public string Type { get; set; }

        public bool IsRead { get; set; }
    }
}

File: mobile/app/src/main/java/com/example/mobile/dto/NotificationDto.kt
package com.example.mobile.dto

/*
This class is used to represent the NotificationDto object that is used to send the notification data to the mobile application.
Notfications are sent to the mobile application to inform the user of any new messages or updates.
*/

data class NotificationDto(
    val id: String,
    val recipientId: String,
    val message: String,
    val messageID: String,
    val role: String,
    val createdAt: String,
    val type: String,
    val isRead: Boolean
)